### DISCUSSION TITLE
ftp from server to Microsoft SharePoint

### DISCUSSION THREAD

**Discussion by paulsally** [iterate-ch/cyberduck#15503] [Q&A]
Hi there.

I tried to move a 22.5 GB zip file from our server directly to SharePoint. It did transferred the file but afterwards it got stuck on merging, I guess. The first time I interrupted it because it seemed to me to be taking an unusual long time. Then I read the documentation and read that the merging phase can take a considerable amount of time.

So I repeated the whole process thinking of leaving it doing it for ever once transfer would end. Again it transferred the file (it reached 100%) but this time I received an Error: server closed connection. Is it worth trying again or is it just systematic?

My connection is not too fast, I have a 10 Mbps network, it was transferring at about 1 MB / s. 
![Screenshot 2024-01-01 at 22 40 50](https://github.com/iterate-ch/cyberduck/assets/46979178/3115d1c0-91a3-42d9-985a-f34a94854ad0)

Cyberduck is set to segment the files.

![Screenshot 2024-01-01 at 04 13 04](https://github.com/iterate-ch/cyberduck/assets/46979178/6402ff21-725e-41ca-b497-55b649a09bb6)

I'd appreciate any help in understanding the situation and how to positively transfer the file.

Also, when I tried again I received an insufficient disk space error. Is it using the disk space in my computer to transfer the files? Where is this temporary file store in my computer so that I can manually delete it?

Thanks a lot.
.S.

**Comment by AliveDevil** [comment]
Theoretically this should work, and not use any disk space.
Issue with large files is, that while being transferred there might be network or server interruptions, that could interfere here.

Observation on my part:
- Dragging a file over to OneDrive/Sharepoint from any remote retrieves _all data_ immediately (that is: a 80 MiB file is tranferred with 100 MiB/s immediately (local source), and ends at 100%) - suggesting, that Cyberduck temporarily stores those files locally.
- The following minutes are kept waiting for the upload to finish (for you, that estimate would be roughly 7 hours, assuming connection speed is constant).

**Reply by AliveDevil** [reply]
I'm not sure, whether, when all data is retrieved locally, and the FTP connection drops due to timeout, the transfer would continue.

**Reply by paulsally** [reply]
I tried again leaving it all night and received this error:
![Screenshot 2024-01-02 at 16 27 47](https://github.com/iterate-ch/cyberduck/assets/46979178/4a9696a6-4bed-453d-bf71-8483b22418e3)
You can see, below the dialog popup, that the data was 100% transferred.

The times I could monitor it I can say it stayed a long time after transfer complete to 100% doing something which I think it's merging all files, the connection didn't seem to drop straightaway but it just can't get to the end...

**Reply by AliveDevil** [reply]
There are two steps in the transfer process here:
- erroneously caching data locally
- uploading that cached data

The 100% mark is what Cyberduck downloaded to local temporary storage.
After completing the download Cyberduck starts uploading the file, without resetting the process as it is a post-transfer action, without progress reporting.

**Reply by paulsally** [reply]
Hi @AliveDevil ,

Thanks for your reply.

I tried to uncheck the "Segmented downloads..." option but I still got an error.  Not sure if the change took effect or not. What else could I try?

Thanks!
.S.

**Reply by AliveDevil** [reply]
You could temporarily store locally, then upload after that - verifying that the data you obtained is good.
Download from FTP to e.g. Downloads, then upload from Downloads to SharePoint, thus you reduce the risk of Cyberduck terminating the transfer due to the FTP server timing out the connection.

**Comment by paulsally** [comment]
Thanks very much Jöran,

I'll try that. Initially I wanted to avoid downloading and then uploading thinking that it was faster to pass from server to server, but I guess it's still downloading and then uploading.

Thanks.

Best regards and Happy New Year.
.S.
